services/sso/src/controllers/dashboard/auth/login.ts

```diff
@@ -12,14 +12,12 @@ import { appConfig } from '../../../config'
 export const dashboardLoginController: Middleware = async (ctx: Context) => {
   const { dni, password }: UserLogin = ctx.request.body
   const dniToUpperCase = dni.toUpperCase()
-  const userResult = await client.query(
-    'SELECT id, password, status, role FROM "user" WHERE dni = $1',
-    [dniToUpperCase]
-  )
-  const user = userResult.rows[0] as Pick<
-    User,
-    'id' | 'password' | 'status' | 'role'
-  >
+  const userResult = await db('user')
+    .select('id', 'password', 'status', 'role')
+    .where('dni', dniToUpperCase)
+    .first()
+
+  const user = userResult as Pick<User, 'id' | 'password' | 'status' | 'role'>
```

Comment:
```suggestion
  const user = await db('user')<User>
    .select('id', 'password', 'status', 'role')
    .where('dni', dniToUpperCase)
    .first()
```
You can infer the type this way as well, [Knex.js documentation](https://knexjs.org/guide/query-builder.html#usage-with-typescript-1)